formats/protobuf/commonMain/src/kotlinx/serialization/protobuf/ProtoBuf.kt

```diff
@@ -135,10 +136,11 @@ class ProtoBuf(
         @Suppress("UNCHECKED_CAST", "NAME_SHADOWING")
         override fun <T> encodeSerializableValue(serializer: SerializationStrategy<T>, value: T) = when {
             // encode maps as collection of map entries, not merged collection of key-values
-            serializer.descriptor is MapLikeDescriptor -> {
+            serializer is MapLikeSerializer<*, *, *, *> -> {
+                // TODO it probably requires some common interface exposure
```

Comment:
I will address it in a follow-up PR

Reply:
We can't, because we have to retrieve key and value serializers.
My plan is to add a simple interface and recommend using it, but not as part of this PR.